### PR TITLE
Don't allow items in the cart that the user cannot purchase (or are deleted) #4539

### PR DESCRIPTION
**Pull request by cklosowski**
#4539

### CODE REVIEW

File: includes/download-functions.php [hunk 1]
```diff
@@ -1239,3 +1239,25 @@ function edd_validate_url_token( $url = '' ) {
...
 }
+
+/**
+ * Checks if the current user can purchase the specified download ID
+ *
+ * @since  2.6.4
+ * @param  integer $download_id The Download ID
+ * @return bool                 If the current user can purcahse the download ID
+ */
+function edd_user_can_purcahse_download( $download_id = 0 ) {
```
Comment (pippinsplugins) [inline includes/download-functions.php]: Can we please make a method of `EDD_Download` please? Also `purchase` is typoed in function name as `purcahse`

**Comment by cklosowski** [comment]
Updated @pippinsplugins

**Comment by pippinsplugins** [comment]
@cklosowski looks like tests are failing. Let's also please add a test that adds a `__return_false` via the filter.

**Comment by cklosowski** [comment]
Failure is unrelated.

The command "bash bin/install-wp-tests.sh wordpress_test root '' localhost
$WP_VERSION" failed and exited with 35 during .

**Comment by pippinsplugins** [comment]
Ah yep, I didn't look at the actual failure. Let's still please add the additional test.

**Pull request merged**